File: solutions/1598. Crawler Log Folder/1598.cpp
// Time complexity: O(n)
// Space complexity: O(1)
class Solution {
 public:
  int minOperations(const vector<string>& logs) {
    int ans = 0;

    for (const string& log : logs) {
      if (log == "./")
        continue;
      if (log == "../")
        ans = max(0, ans - 1);
      else
        ++ans;
    }

    return ans;
  }
};

File: solutions/1598. Crawler Log Folder/1598.py
# Time complexity: O(n)
# Space complexity: O(1)
class Solution:
  def minOperations(self, logs: List[str]) -> int:
    return reduce(
      lambda acc, log: max(0, acc - 1) if log == "../" else
                       acc + 1 if log != "./" else acc,
      logs,
      0
    )